flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -206,4 +287,9 @@ private DescribeStreamResult describeStream(String streamName, String startShard
 			throw new RuntimeException("Stream is not Active or Updating");
```

Comment:
Actually, should we really be throwing a RuntimeException if a single stream is not in ACTIVE or UPDATING status? Say the consumer is to fetch 3 streams, and only 1 is found to be CREATING / DELETING. Perhaps we should treat this case the same as "can only find shards for some of the streams", and a warning log here will do?

There's a check that stops the consumer in the new `ShardDiscoverer` if no shards can be initially found at all to read from anyway.